core/src/main/java/org/apache/brooklyn/core/objs/BasicSpecParameter.java

```diff
@@ -247,17 +248,17 @@ public String toString() {
                 .name(name)
                 .description(description)
                 .defaultValue(defaultValue)
-                .constraint(constraints)
+                .constraints(constraints)
                 .runtimeInheritance(parentInheritance)
                 .typeInheritance(typeInheritance);
-            
+
             if (PortRange.class.equals(typeToken.getRawType())) {
                 sensorType = new PortAttributeSensorAndConfigKey(builder);
                 configType = ((HasConfigKey)sensorType).getConfigKey();
             } else {
                 configType = builder.build();
             }
-            return new BasicSpecParameter(Objects.firstNonNull(label, name), true, configType, sensorType);
+            return new BasicSpecParameter(Objects.firstNonNull(label, name), Objects.firstNonNull(pinned, Boolean.FALSE), configType, sensorType);
```

Comment:
The usual way to express this in the codebase is: `Boolean.TRUE.equals(pinned)`, but this works as well.